api_tests/dry/config.ts

```diff
@@ -0,0 +1,83 @@
+import { execSync, exec } from "child_process";
+import { promisify } from "util";
+import { expect } from "chai";
+import tmp from "tmp";
+import * as fs from "fs";
+
+const execAwait = promisify(exec);
+
+const projectRoot: string = execSync("git rev-parse --show-toplevel", {
+    encoding: "utf8",
+}).trim();
+
+const defaultConfigComplete = `[network]
+listen = ["/ip4/0.0.0.0/tcp/9939"]
+[http_api.socket]
+address = "0.0.0.0"
+port = 8000
+
+[http_api.cors]
+allowed_origins = "none"
+
+[data]
+dir = "/Users/bonomat/Library/Application Support/comit"
```

Comment:
Username and OSX specific path here mate.

Reply:
yeah that is going to fail on every other machine :sweat_smile:.

Not sure how to solve that best. so I am gonna just ask a few questions:

What exactly is the goal of this tests?

1. Testing that `--dump-config` exists?
2. Testing whether the output of `--dump-config` is a valid config? It is a good idea to do so, we currently lack tests for our CLI interface. But for that, it would be good enough to treat the config as opaque, write it to a temp file and test if cnd starts up correctly with its own defaults.
3. Testing the default values of the config file? We do that in Rust already, not sure if we want to test this level of detail on the outside, the problem with OS and user specific paths stems from doing that.

Reply:
> yeah that is going to fail on every other machine 😅.

I'm going to force everyone to create this folder or e2e tests will fail on your machine 
(PR to change config generation in cnd will follow) 🤪

> What exactly is the goal of this tests?

I realized that we don't test `--dump-config` at all. 
The goal was to test that cnd is able to dump a complete config file. It was only afterwards that I realized that we have extensive tests in rust to test that. 

I'll remove this test again as it does not add that much value. 

@thomaseizinger , regarding your point 2:
I think this would not test if cnd generated a complete config file.

Reply:
>  I think this would not test if cnd generated a complete config file.

True, but I also didn't claim that :)

Reply:
To test that, I would write a test in TS that parses the toml and verifies that certain sections are there.